src/engine/execute.js

```diff
@@ -19,35 +19,33 @@ var execute = function (sequencer, thread) {
     var runtime = sequencer.runtime;
     var target = thread.target;
 
-    // Current block to execute is the one on the top of the stack.
-    var currentBlockId = thread.peekStack();
-    var currentStackFrame = thread.peekStackFrame();
-
-    // Check where the block lives: target blocks or flyout blocks.
-    var targetHasBlock = (
-        typeof target.blocks.getBlock(currentBlockId) !== 'undefined'
-    );
-    var flyoutHasBlock = (
-        typeof runtime.flyoutBlocks.getBlock(currentBlockId) !== 'undefined'
-    );
-
     // Stop if block or target no longer exists.
-    if (!target || (!targetHasBlock && !flyoutHasBlock)) {
+    if (target === null) {
```

Comment:
I'm fine with this new code as-is but I find it interesting that the old code accessed `target` before this line, in the `targetHasBlock` test. I guess this branch isn't used very often in practice :)